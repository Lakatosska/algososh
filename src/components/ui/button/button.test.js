import renderer from 'react-test-renderer';
import {render, screen, fireEvent} from '@testing-library/react';
import { Button } from './button';

describe('Тестирование компонента Button', () => {

  it('Кнопка с текстом', () => {
    const button = renderer.create(<Button text='text' />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Кнопка без текста', () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Заблокированная кнопка', () => {
    const button = renderer.create(<Button disabled={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Кнопка с индикацией загрузки', () => {
    const button = renderer.create(<Button isLoader={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Вызов колбэка при клике на кнопку', () => {
    const mockFn = jest.fn();
    render(<Button text='text' onClick={mockFn} />);
    const button = screen.getByText('text');
    fireEvent.click(button);
    expect(mockFn).toHaveBeenCalled();
  })

})