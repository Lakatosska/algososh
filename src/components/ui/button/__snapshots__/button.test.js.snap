// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Button Заблокированная кнопка 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button Кнопка без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button Кнопка с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование компонента Button Кнопка с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;
